web/client/components/mapcontrols/searchcoordinates/CurrentMapCRSCoordSearch.js

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2024, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+*/
+
+import React from 'react';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+
+import {FormGroup, InputGroup, Row} from "react-bootstrap";
+
+import Message from "../../I18N/Message";
+import CoordinateEntry from "../../misc/coordinateeditors/CoordinateEntry";
+import { zoomAndAddPoint, changeCoord } from '../../../actions/search';
+import {getExtentByCRSCode, reproject} from '../../../utils/CoordinatesUtils';
+import {CoordinateOptions} from './CoordinatesSearch';
+/**
+ * CurrentMapCRSCoordinatesSearch component
+ * @param {object} props Component props
+ * @param {object} props.coordinate coordinate position in x y
+ * @param {string} props.format current format to display the coordinates
+ * @param {func} props.onClearCoordinatesSearch on click clear coordinates
+ * @param {func} props.onZoomToPoint on click zoom to extent
+ * @param {number} props.defaultZoomLevel default zoom level on zoom to extent
+ * @param {func} props.onChangeCoord triggered on changing coordinate position
+ *
+ */
+
+const CurrentMapCRSCoordinatesSearch = ({
+    coordinate = {},
+    format,
+    onClearCoordinatesSearch,
+    onZoomToPoint,
+    onChangeCoord,
+    defaultZoomLevel,
+    currentMapCRS
+}) => {
+    const {zoomToPoint, areValidCoordinates} = CoordinateOptions;
+    React.useEffect(() => {
+        if (coordinate.lat) {
+            const latValue = coordinate.lat ? reproject([0, coordinate.lat], 'EPSG:4326', currentMapCRS, true)?.y : undefined;
+            onChangeCoord('yCoord', latValue);
+        }
+        if (coordinate.lon) {
+            const lonValue = coordinate.lon ? reproject([coordinate.lon, 0], 'EPSG:4326', currentMapCRS, true)?.x : undefined;
+            onChangeCoord('xCoord', lonValue);
+        }
+    }, []);
+
+    const getConstraintsCoordEditor = (projection) =>{
+        const [minx, miny, maxx, maxy] = getExtentByCRSCode(projection);
+        return {
+            decimal: {
+                yCoord: {
+                    min: miny,
+                    max: maxy
+                },
+                xCoord: {
+                    min: minx,
+                    max: maxx
+                }
+            }
+        };
+    };
+    const changeCoordinates = (coord, value) => {
+        const numValue = parseFloat(value);
+        onChangeCoord(coord, numValue);
+        if (coord === 'yCoord') {
+            const latValue = numValue ? reproject([0, numValue], currentMapCRS, 'EPSG:4326', true)?.y : undefined;
+            onChangeCoord('lat', latValue ? Number((latValue).toFixed(4)) : latValue);
```

Comment:
I don't think the value should use fixed decimal places. Kindly refer to coordinate search if there is any similar limits applied